### DISCUSSION TITLE
SX1262-Implement Retransmission Mechanism

### DISCUSSION THREAD

**Discussion by Nandhini-271** [jgromes/RadioLib#1198] [General]
I'm using the SX1262 LoRa module with an ESP32, programmed using the RadioLib library. This library has a built-in retransmission mechanism, where if the `startTransmit` or `startReceive` function fails, it automatically attempts retransmission. Could you please provide clarity on how this retransmission process works?

**Comment by jgromes** [comment]
> This library has a built-in retransmission mechanism

No, it does not. There's no such functionality directly in RadioLib. If you need it, you will have to either find a higher-layer protocol that does have it, or implement it yourself.

**Reply by Nandhini-271** [reply]
Which higher level protocol is used for retransmission?

**Reply by StevenCellist** [reply]
LoRaWAN is a protocol that implements frame retransmission, but this is not implemented in the current release of RadioLib. I do have it implemented in a local copy that is likely to see the daylight through a PR in a week or maybe two. But whether you can use LoRaWAN highly depends on your situation, options & requirements.
